akka-diagnostics/src/main/scala/akka/diagnostics/StarvationDetector.scala

```diff
@@ -133,21 +133,22 @@ object StarvationDetector {
    * Creates and runs a StarvationDetector thread for the dispatcher of the system's main dispatcher, i.e.
    * akka.actor.default-dispatcher.
    */
-  def checkSystemDispatcher(system: ActorSystem): Unit =
+  def checkSystemDispatcher(provider: ClassicActorSystemProvider): Unit =
     checkSystemDispatcher(
-      system,
-      StarvationDetectorSettings.fromConfig(system.settings.config.getConfig("akka.diagnostics.starvation-detector")))
+      provider,
+      StarvationDetectorSettings.fromConfig(
+        provider.classicSystem.settings.config.getConfig("akka.diagnostics.starvation-detector")))
 
   /**
    * Creates and runs a StarvationDetector thread for the dispatcher of the system's main dispatcher, i.e.
    * akka.actor.default-dispatcher, with custom configuration.
    */
-  def checkSystemDispatcher(system: ActorSystem, config: StarvationDetectorSettings): Unit =
+  def checkSystemDispatcher(provider: ClassicActorSystemProvider, config: StarvationDetectorSettings): Unit =
     checkExecutionContext(
-      system.dispatcher,
-      Logging(system, classOf[StarvationDetectorThread]),
+      provider.classicSystem.dispatcher,
+      Logging(provider.classicSystem, classOf[StarvationDetectorThread]),
       config,
-      () => system.whenTerminated.isCompleted)
+      () => provider.classicSystem.whenTerminated.isCompleted)
 
```

Comment:
we also have in ConfigChecker
```
def reportIssues(system: ExtendedActorSystem)
```

That can also be ClassicActorSystemProvider. I don't know if we need the Extended, but if that is the case you can safely cast it `provider.classicSystem.asInstanceOf[ExtendedActorSystem]`

Reply:
Okay. I'll do that too.